File: problems/1684 - Count the Number of Consistent Strings/Kotlin/Solution.kt
class Solution {
    fun countConsistentStrings(allowed: String, words: Array<String>): Int {
        var count = words.size
        val wordSet = IntArray(26)

        for (i in allowed.indices) {
            wordSet[allowed[i] - 'a']++
        }

        for (word in words) {
            for (i in word.indices) {
                if (wordSet[word[i] - 'a'] == 0) {
                    count--
                    break
                }
            }
        }

        return count
    }
}

File: problems/1684 - Count the Number of Consistent Strings/PHP/Solution.php
<?php

class Solution {

    /**
     * @param String $allowed
     * @param String[] $words
     * @return Integer
     */
    function countConsistentStrings($allowed, $words) {
        $count = count($words);
        $wordSet = array_fill(0, 26, 0);

        for ($i = 0; $i < strlen($allowed); $i++) {
            $wordSet[ord($allowed[$i]) - ord('a')]++;
        }

        foreach ($words as $word) {
            for ($i = 0; $i < strlen($word); $i++) {
                if ($wordSet[ord($word[$i]) - ord('a')] == 0) {
                    $count--;
                    break;
                }
            }
        }

        return $count;
    }
}